### ISSUE TITLE
Flutter and Phoenix channels

### ISSUE THREAD

**Issue by th0mas** [dwyl/learn-flutter#61]
Flutter/Dart currently only has one Phoenix client library so we have two approaches to integrate Flutter and Phoenix:

+ Use the existing library: https://github.com/mfeckie/phoenix_wings
  This is probably the easiest option, but we will run into some bugs/edge cases, such as heartbeat `ack`s that aren't 
  implemented by this library so we may have to make changes anyway.

+ Run our own on top of Darts [web_socket_channel](https://pub.dev/packages/web_socket_channel)
  phoenix.js `cloc`s in at around 800 lines of OO JavaScript code, so it wouldn't be a huge task to port this over to dart.
  This would provide us with a consistent API with the official library. I'm happy to do this as a "learn dart" exercise

**Comment by nelsonic** [comment]
@th0mas see how far you can get with **`phoenix_wings`** and leave comments on this issue with your progress.
If you hit a brick wall, please open an issue on https://github.com/mfeckie/phoenix_wings/issues and link to it from here.
Groking the source will be informative if you decide to re-implement it.

**Comment by nelsonic** [comment]
Crucially we need to have a library that knows to avoid attempting to reconnect when the devices is offline 
and automatically reconnects when there is a connection, so it may need to be _aware_ of connection status #3.
Also, having a _heartbeat_ could be useful for presence https://github.com/mfeckie/phoenix_wings/issues/10 as we want to know "_Who's Online?_"

**Comment by th0mas** [comment]
This library does seem to have Presence support (https://pub.dev/documentation/phoenix_wings/latest/html/PhoenixPresence-class.html) so that shouldn't be a problem.

I'll check for online/offline callbacks.

**Comment by SimonLab** [comment]
Just had a look at the `phoenix_wings` repository, the last updated was a while ago now:
![image](https://user-images.githubusercontent.com/6057298/215741245-44986902-a23d-4e1c-b3c6-1e1f22305794.png)

With the newest version of Phoenix I'm not sure if this library is still compatible.

**Comment by nelsonic** [comment]
@SimonLab yeah, the top result when searching for [google.com/search?q=flutter+phoenix+channels](https://www.google.com/search?q=flutter+phoenix+channels)
Is no longer `phoenix_wings` ...

<img width="1276" alt="google-phoenix-flutter-channels" src="https://user-images.githubusercontent.com/194400/215790292-e7433d92-ff1f-4682-9a6f-24d8f1a1d2a9.png">

It's `phoenix_socket`: https://pub.dev/packages/phoenix_socket
<img width="1226" alt="image" src="https://user-images.githubusercontent.com/194400/215789702-2028e906-7eb0-47b7-a15e-314b580b5e26.png">

Which has recent activity/contribution: https://github.com/braverhealth/phoenix-socket-dart
<img width="1226" alt="image" src="https://user-images.githubusercontent.com/194400/215790497-1adef9ec-555a-4524-b03a-1028942a75a0.png">

It has a "corporate sponsor" in the form of: https://braver.net/en/
<img width="1261" alt="image" src="https://user-images.githubusercontent.com/194400/215791059-b678c5d0-093d-42cd-867f-dc77808784f3.png">


And the lead dev @matehat seems to know what he's doing. 👌 
<img width="1343" alt="image" src="https://user-images.githubusercontent.com/194400/215791269-1271c6c4-ef1c-4b05-86c5-bc1c81322a0f.png">

This is the approach/route I would take. 💭

**Comment by nelsonic** [comment]
Quick issue https://github.com/braverhealth/phoenix-socket-dart/issues/ 
and PR to test https://github.com/braverhealth/phoenix-socket-dart/pull/57 receptiveness to future contributions. 😉

**Comment by nesimtunc** [comment]
Is there any issue? I wonder if anyone using it in production one of those libraries?
